### ISSUE TITLE
No checks on registered resource providers

### ISSUE THREAD

**Issue by michaeljcameron** [MicrosoftDocs/mslearn-aspnet-core#89]
The setup scripts appear to enter an infinite loop if required resource providers are not registered prior to running.  If this is your first venture into running AKS in your subscription that will be quite likely given that this is an introductory module.
I'm not 100% certain about this but there are errors in the activity log under "Create or Update Managed Cluster"

`Operation name
Create or Update Managed Cluster
Time stamp
Thu Jun 03 2021 08:00:29 GMT+0100 (British Summer Time)
Event initiated by
myaddress@email.com
Error code
MissingSubscriptionRegistration
Message
The subscription is not registered to use namespace 'Microsoft.ContainerService'. See https://aka.ms/rps-not-found for how to register subscriptions.`

This is the script output.  Yes, it does take a long time to get the LB IP address so the error might not result in infinite loop.
Will test if I get more time

` > az network public-ip list -g MC_eshop-learn-rg_eshop-learn-aks_westus --query "[?tags.service=='ingress-nginx/ingress-nginx'].ipAddress" -otsv

Waiting for the load balancer IP address...

 > az network public-ip list -g MC_eshop-learn-rg_eshop-learn-aks_westus --query "[?tags.service=='ingress-nginx/ingress-nginx'].ipAddress" -otsv

Waiting for the load balancer IP address...

 > az network public-ip list -g MC_eshop-learn-rg_eshop-learn-aks_westus --query "[?tags.service=='ingress-nginx/ingress-nginx'].ipAddress" -otsv

Waiting for the load balancer IP address...

 > az network public-ip list -g MC_eshop-learn-rg_eshop-learn-aks_westus --query "[?tags.service=='ingress-nginx/ingress-nginx'].ipAddress" -otsv

Waiting for the load balancer IP address...

 > az network public-ip list -g MC_eshop-learn-rg_eshop-learn-aks_westus --query "[?tags.service=='ingress-nginx/ingress-nginx'].ipAddress" -otsv`

**Comment by CamSoper** [comment]
@michaeljcameron Is this a personal subscription or an enterprise subscription? There might be permissions issues we need to work around.

**Comment by CamSoper** [comment]
Oh, and thank you for taking the time to log an issue! Sorry it's not working for you. ☹️ Which Learn module are you trying to get through? The microservice introductory one? We use this script across several.

**Comment by michaeljcameron** [comment]
I've returned to this to see if I can replicate it again easily.

I ran these commands:
```
az provider unregister --namespace Microsoft.ContainerRegistry
az provider unregister --namespace Microsoft.ContainerInstance
az provider unregister --namespace Microsoft.ContainerService
```
then reran the setup
`. <(wget -q -O - https://aka.ms/microservices-aspnet-core-setup)
`

It ran for a loooonnnnggggg time, about three quarters of an hour but it did succeed (with an error):

`The subscription registration is in 'Unregistered' state. The subscription must be registered to use namespace 'Microsoft.ContainerRegistry'. See https://aka.ms/rps-not-found for how to register subscriptions.`

I noted after that some providers which I had removed were reinstalled so probably a "user misunderstanding, close no change"
